src/Maui/Prism.Maui/Controls/PrismNavigationPage.cs

```diff
@@ -39,4 +41,17 @@ private async void HandleBackButtonPressed(object sender, EventArgs args)
     {
         await MvvmHelpers.HandleNavigationPageGoBack(this);
     }
+
+    protected override async void OnDisappearing()
+    {
+        var presentationStyle = Microsoft.Maui.Controls.PlatformConfiguration.iOSSpecific.Page.GetModalPresentationStyle(this);
+
+        if (presentationStyle != UIModalPresentationStyle.FullScreen && PageNavigationService.NavigationSource == PageNavigationSource.Device)
+        {
+            await MvvmHelpers.HandleNavigationPageSwipedAway(this);
+        }
+
+        base.OnDisappearing();
+    }
```

Comment:
please wrap this in a compiler directive and update the csproj to target net8.0-ios... does this also need to be done for maccatalyst?

Reply:
I've added `net8.0-ios` as a target and wrapped in a compiler directive.

I've had a look at the mac catalyst implementation but the popover presentation styles can't be automatically dismissed so we don't run into the same issue faced by iOS